### ISSUE TITLE
Create Output.all alternative as an object / dict

### ISSUE THREAD

**Issue by colearendt** [pulumi/pulumi#6003]
`Output.all()` (in Python) creates a list. It would be great have some variant of `Output.all()` that creates an object or dict so keys / objects are easier to reference by name!

Slack thread: https://pulumi-community.slack.com/archives/C84L4E3N1/p1608681663484600

## Expected Behavior
```
# something like
Output.all(name=output1, name2=output2).apply(lambda x: x['name'])
```

## Current Behavior
```
# instead of
Output.all(output1, output2).apply(lambda x: x[0])
```

## Steps to Reproduce
N/A

## Context (Environment)

I have helper functions that do things like **create_a_vpc**. There are many outputs here, so it is nice returning `Output.all(output1, output2, output3)`. However, this means utilizing those resources later need to be referenced by index rather than by name.

I'm not sure whether IDEs would be able to pick up the keys for the `Output` type of objects, but if so, autocomplete would be amazing here as well!!

**Comment by nbyl** [comment]
This would greatly help us. We have a pretty neat use case for that. We have an application that needs a yaml config file that must be written to a Kubernetes secret. The yaml structure contains values that are based on outputs from other resources (database, buckets, ...). This get pretty confusing if you have more than just a handful parameters.

**Comment by komalali** [comment]
This is included in [v2.21.0](https://pypi.org/project/pulumi/) of the SDK.

It is now possible to do the following:
```python
Output.all(name=output1, name2=output2).apply(lambda x: x['name'])
```
